src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -270,6 +270,14 @@ public void StartConnectAsync()
                     peer?.Disconnect("Connection timeout");
                 }
             }
+
+            catch (NBitcoin.Protocol.ProtocolException)
+            {
+                this.logger.LogDebug("Peer {0} handshake failed.", peerAddress.Endpoint);
```

Comment:
I think so, there are several ways how the handshake can fail in this one is triggered only when our version message is rejected by the other side, so it might be slightly better to say something like
```
this.logger.LogDebug("Handshake rejected by peer '{0}'.", peerAddress.Endpoint);
```

Reply:
Done.